snuba/subscriptions/store.py

```diff
@@ -0,0 +1,104 @@
+import json
+from dataclasses import dataclass
+from datetime import timedelta
+from typing import Collection
+
+from rediscluster import StrictRedisCluster
+
+from snuba.subscriptions.data import Subscription
+from snuba.utils.codecs import Codec
+
+
+@dataclass(frozen=True)
+class RedisPayload:
+    __slots__ = ["key", "value"]
+
+    key: str
+    value: bytes
+
+
+class SubscriptionCodec(Codec[RedisPayload, Subscription]):
+    def encode(self, value: Subscription) -> RedisPayload:
+        return RedisPayload(
+            value.id,
+            json.dumps(
+                {
+                    "id": value.id,
+                    "project_id": value.project_id,
+                    "conditions": value.conditions,
+                    "aggregations": value.aggregations,
+                    "time_window": int(value.time_window.total_seconds()),
+                    "resolution": int(value.resolution.total_seconds()),
+                }
+            ).encode("utf-8"),
+        )
+
+    def decode(self, value: RedisPayload) -> Subscription:
+        data = json.loads(value.value.decode("utf-8"))
+        return Subscription(
+            id=data["id"],
+            project_id=data["project_id"],
+            conditions=data["conditions"],
+            aggregations=data["aggregations"],
+            time_window=timedelta(seconds=data["time_window"]),
+            resolution=timedelta(seconds=data["resolution"]),
+        )
+
+
+class SubscriptionDoesNotExist(Exception):
+    pass
+
+
+class RedisSubscriptionStore:
```

Comment:
Is this supposed to have the same interface of the Kafka store ? Or is it meant to be a different concept ? If it is supposed to be the same, could we have the abstract class to begin with so we can catch incompatibilities in the two interfaces early ?

Reply:
We'll have a Kafka store for these at some point soon, and the Redis store will be more like a long lived cache, with Kafka acting kind of like a WAL. I'm not totally sure what the interface will look like yet, and so punted on designing the abstraction until we know how the Kafka piece will work.